linux/LinuxProcessTable.c

```diff
@@ -1098,6 +1098,33 @@ static void LinuxProcessTable_readCwd(LinuxProcess* process, openat_arg_t procFd
    free_and_xStrdup(&process->super.procCwd, pathBuffer);
 }
 
+/*
+ * Read /proc/<pid>/syscall (thread-specific data)
+ */
+static void LinuxProcessTable_readSyscall(LinuxProcess* process, openat_arg_t procFd) {
+   char buffer[PATH_MAX];
+   ssize_t r;
+
+   r = xReadfileat(procFd, "syscall", buffer, sizeof(buffer));
+   if (r <= 0) {
+      free(process->syscall);
+      process->syscall = NULL;
+      return;
+   }
+
+   /* Remove '\n' */
+   buffer[r - 1] = '\0';
```

Comment:
This should use `strchrnul` to remove the `\n` optionally, in case it doesn't exist.

Also no need to trim `\n` if we already split at any other whitespace. What about using `fast_strtoull` and check for proper conversion? As we skip arguments anyway this should suffice and would additionally allow proper display as string (syscall name) later.

Reply:
Task might not have doing any syscall, which is "running" status in the file, so I think currently we using strings is the most appropriate.

Reply:
@JohnSanpe You should handle the `running` string as special and assign a placeholder number for it. The reason is it allows us to display `running` string with colors. This status should be special in sorting/collating anyway.